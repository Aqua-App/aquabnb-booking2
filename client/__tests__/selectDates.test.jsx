import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import SelectDates from '../components/selectDates';
import Adapter from 'enzyme-adapter-react-16';

// import { configure } from "enzyme";
// import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

describe('Test suite for selectDates component', () => {

  it('it renders without crashing', () => {
    const wrapper = shallow(<SelectDates />);
    expect(wrapper).toBeDefined();
  })

  it('checks that Popup is rendered checkin click', () => {
    const wrapper = shallow(<SelectDates />);
    expect(wrapper.find('Popup').exists()).toBeFalsy();
    wrapper.find('#in').simulate('click');
    expect(wrapper.find('Popup').exists()).toBeTruthy();
  })

  it('checks that Popup is rendered checkout click', () => {
    const wrapper = shallow(<SelectDates />);
    expect(wrapper.find('Popup').exists()).toBeFalsy();
    wrapper.find('#out').simulate('click');
    expect(wrapper.find('Popup').exists()).toBeTruthy();
  })

})

