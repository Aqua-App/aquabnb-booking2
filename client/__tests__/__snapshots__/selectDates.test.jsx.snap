// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for selectDates component should match snapshot 1`] = `
<styled.div>
  <styled.div
    id="in"
    onClick={[Function]}
  >
    <b>
      CHECK-IN
    </b>
    <div
      style={
        Object {
          "color": "#717171",
          "font-size": "14px",
        }
      }
    />
  </styled.div>
  <styled.div
    id="out"
    onClick={[Function]}
  >
    <b>
      CHECK-OUT
    </b>
    <div
      style={
        Object {
          "color": "#717171",
          "font-size": "14px",
        }
      }
    />
  </styled.div>
  <GuestSelect />
</styled.div>
`;
